crates/motoko/src/lib/vm.rs

```diff
@@ -203,32 +204,33 @@ fn call_prim_function(
     core: &mut Core,
     pf: &PrimFunction,
     targs: Option<Inst>,
-    args: Value,
+    args: Value_,
 ) -> Result<Step, Interruption> {
     use PrimFunction::*;
     match pf {
-        DebugPrint => match args {
+        DebugPrint => match &*args {
             Value::Text(s) => {
                 log::info!("DebugPrint: {}: {:?}", core.cont_source, s);
-                core.debug_print_out.push_back(s);
-                core.cont = Cont::Value(Value::Unit);
+                core.debug_print_out.push_back(s.clone()); // TODO: store debug output as `Value_`?
```

Comment:
Yeah, I think that makes sense.  That way, we can share the same text value there.